sympy/physics/mechanics/tests/test_functions.py

```diff
@@ -250,3 +252,76 @@ def test_center_of_mass():
     point_o.set_pos(p1.point, center_of_mass(p1.point, p1, p2, p3, p4, b))
     expr = 5/(m + mb + 6)*a.x + (m + mb + 3)/(m + mb + 6)*a.y + mb/(m + mb + 6)*a.z
     assert point_o.pos_from(p1.point)-expr == 0
+
+def test_apply_torque():
+    t = symbols('t')
+    q = dynamicsymbols('q')
+    B1 = Body('B1')
+    B2 = Body('B2')
+    N = ReferenceFrame('N')
+    torque = t*q*N.x
+
+    apply_torque(torque, B1, B2) #Applying equal and opposite torque
+    assert B1.loads == [(B1.frame, torque)]
+    assert B2.loads == [(B2.frame, -torque)]
+
+    torque2 = t*N.y
+    apply_torque(torque2, B1)
+    assert B1.loads == [(B1.frame, torque+torque2)]
+
+def test_apply_force():
+    f, g = symbols('f g')
+    q, x, v1, v2 = dynamicsymbols('q x v1 v2')
+    P1 = Point('P1')
+    P2 = Point('P2')
+    B1 = Body('B1')
+    B2 = Body('B2')
+    N = ReferenceFrame('N')
+
+    P1.set_vel(B1.frame, v1*B1.x)
+    P2.set_vel(B2.frame, v2*B2.x)
+    force = f*q*N.z # time varying force
+
+    apply_force(force, B1, B2, P1, P2) #applying equal and opposite force on moving points
+    assert B1.loads == [(P1, force)]
+    assert B2.loads == [(P2, -force)]
+
+    g1 = B1.mass*g*N.y
+    g2 = B2.mass*g*N.y
+
+    apply_force(g1, B1) #applying gravity on B1 masscenter
+    apply_force(g2, B2) #applying gravity on B2 masscenter
+
+    assert B1.loads == [(P1,force), (B1.masscenter, g1)]
+    assert B2.loads == [(P2, -force), (B2.masscenter, g2)]
+
+    force2 = x*N.x
+
+    apply_force(force2, B1, B2) #Applying time varying force on masscenter
+
+    assert B1.loads == [(P1, force), (B1.masscenter, force2+g1)]
+    assert B2.loads == [(P2, -force), (B2.masscenter, -force2+g2)]
+
+def test_apply_loads_on_multi_degree_freedom_holonomic_system():
+    W = Body('W') #Wall
+    B = Body('B') #Block
+    P = Body('P') #Pendulum
+    b = Body('b') #bob
+    q1, q2 = dynamicsymbols('q1 q2') #generalized coordinates
+    k, c, g, kT = symbols('k c g kT') #constants
+    F, T = dynamicsymbols('F T') #Specified forces
+
+    #Applying forces
+    B.apply_force(F*W.x)
```

Comment:
What is the purpose of having and `apply_force()` method on `Body` and a function called `apply_force`? Why can't, for example, the method do everything the function can?

Reply:
I was thinking `apply_force()` could be used to apply equal and opposite forces with added functionality of single forces, whereas `Body.apply_force` could be used to add single forces. Combining them might make it a bit complicated as it was in orient().

Reply:
The function `apply_force()` already has a certain level of complexity and you have an identical method on Body that doesn't do the same thing. I'm not seeing why we need the function if we have the method? Here are scenarios:

1. User has to manually handle equal and opposite forces and `.apply_force()` only accepts a single force.

```python
A = Body('A')
B = Body('B')
A.apply_force(F*A.x)
B.apply_force(-F*A.x)
```

2. Equal and opposite can be handled automatically by `.apply_force()` by `.apply_force()` accepting a body.

```python
A = Body('A')
B = Body('B')
A.apply_force(F*A.x, reaction_body=B)
```

3. Equal and opposite can be handled automatically by `apply_force()` (function).

```python
A = Body('A')
B = Body('B')
apply_force(F*A.x, reaction_body1=A, reaction_body2=B)
```

2. seems the most straight forward with no redundancy to me and less typing, at least from this minimal example. 3. has redundant methods and functions. It isn't clear why having two ways to add the equal & opposite force gives benefit.

If there are example problems that show why the function is better, show a demonstration.

Reply:
And, to be honest, point 1. is pretty clean and simple. It just requires the user to do the correct thing (garbage in, gargage out philosophy).

Reply:
Yes, I told you about way 2 in meeting, but you asked me to proceed with 3.

I will change it to 2 , I also prefer 2.

But there is one other way also, i.e , we can use Body's attribute for single forces and `apply_force` function for equal and opposite forces.

So way 2 or this way that I explained?

Reply:
I recall you suggesting a function not a method in the meeting, but maybe I misunderstood. I seemed reasonable then, but we hadn't seen the examples of its use. Regardless, I don't know what the best option is until we've explored the possibilities. This is the purpose of trying to come up with examples for use and writing them down. We can then examine the proposed API design in the context of the examples.

The process is to:

1. Think of all possible examples (including edge cases) that we want the API to be used for. These examples should be derived from existing problems we've created over the years as well as looking at other similar tools and how they approach the problems.
2. Propose an API design.
3. Show the API design used in the examples in a PR.
4. Discuss and debate whether the API design is good or needs improvement.
5. Modify the API to make it better.
6. Repeat starting at 1. if you think of new examples that you didn't originally think of.

Steps 1 to 6 are repeated until everyone involved in the design is happy with the result.

You job is to do this process and I will help with point 1. by pointing you to examples and providing my past experience and in 4. by discussing your proposal.

For this particular API idea, I'm not yet sure point 1. is fully fleshed out and if it is, then the API is too complex, thus my suggestions.

Reply:
We will discuss it further in upcoming meet.